src/consumer.cpp

```diff
@@ -248,6 +246,24 @@ vector<Message> Consumer::poll_batch(size_t max_batch_size, milliseconds timeout
     return output;
 }
 
+Queue Consumer::get_main_queue() const {
+    Queue queue = Queue::make_non_owning(rd_kafka_queue_get_main(get_handle()));
+    queue.disable_queue_forwarding();
+    return queue;
+}
+
+Queue Consumer::get_consumer_queue() const {
+    return Queue::make_non_owning(rd_kafka_queue_get_consumer(get_handle()));
+}
+
+Queue Consumer::get_partition_queue(const TopicPartition& partition) const {
+    Queue queue = Queue::make_non_owning(rd_kafka_queue_get_partition(get_handle(),
```

Comment:
I thought so too based on the librdkafka documentation (misleading _remove reference_ comment). However based on my initial testing, once you destroy a queue, it's gone forever for the lifetime of the consumer. And since I need to reuse them, I can't destroy them. I will redo this test once again to confirm but unfortunately it's how the library works.

Reply:
Looks like this is worth raising as a bug on the rdkafka repo. I don't think that should be the expected behavior. I'm wondering if this leaks memory as it is though, it seems kind of eerie.